ProcessRateLimitedMessages/index.js

```diff
@@ -14,6 +14,8 @@ const toSendQueueClient = new QueueClient(connectionString, toSendQueue)
 
 async function ensureResourcesExist () {
   await toSendQueueClient.createIfNotExists()
+  await failedToSendQueueClient.createIfNotExists()
+  await batchesContainerClient.createIfNotExists()
```

Comment:
you could run these three `await`s in parallel with a `Promise.all` as you've used elsewhere, but I take it this shouldn't run very often, so perhaps wasn't worth the extra complexity in the code?

Reply:
No, that is a very good point. I'll make that change.
Ideally this check would only run when the application starts up but as that isn't really the way functions work it needs to be run for every execution. I would have moved it to outside of the main function to use the same principle as the azure storage clients i.e. can be shared between functions, however, with no top level awaits that isn't possible.
An alternative is to not make the check and ensure all resources exists during the deployment, however, mistakes can happen during deployment so I'd like to keep a check in to cover that scenario. I've not seen any performance issues with the checks in other functions but would be something to look to change should performance be a problem.

Reply:
Using Promise.all() outside of the function works to have the resources created during the module load rather than waiting for the function execution. Should be the most performant option.